hydra-cardano-api/src/Hydra/Cardano/Api/Tx.hs

```diff
@@ -234,12 +236,13 @@ toLedgerTx = \case
 fromLedgerTx :: Ledger.Tx (ShelleyLedgerEra Era) -> Tx Era
 fromLedgerTx ledgerTx =
   Tx
-    (ShelleyTxBody shelleyBasedEra body scripts scriptsData (strictMaybeToMaybe auxData) validity)
+    (ShelleyTxBody shelleyBasedEra body' scripts scriptsData (strictMaybeToMaybe auxData) validity)
     (fromLedgerTxWitness wits)
  where
   -- XXX: The suggested way (by the ledger team) forward is to use lenses to
   -- introspect ledger transactions.
   Ledger.AlonzoTx body wits isValid auxData = ledgerTx
+  body' = body & auxDataHashTxBodyL .~ (hashTxAuxData <$> auxData)
```

Comment:
This (or above) looks much more likely to be something that helped the root cause..

Reply:
we can test if all works without this commit since it didn't fix the original issue according to MLabs.